src/AgentImplementation.sol

```diff
@@ -280,4 +351,8 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
             }
         }
     }
+
+    function _calculateFee(uint256 feeRate, uint256 amount) internal pure returns (uint256) {
+        return (amount * feeRate) / (_BPS_BASE + feeRate);
```

Comment:
Wondering why we need to add `feeRate` in `_BPS_BASE + feeRate`?

Reply:
hmm good question. Maybe it's needed in the previous implementation but not now. Let me think about that.